gatorgrouper/utils/read_student_file.py

```diff
@@ -13,17 +13,30 @@ def read_student_file(filepath):
 
     # read the raw CSV data
     with open(filepath, "r") as csvfile:
-        csvdata = list(csv.reader(csvfile, delimiter=","))
+        has_header = csv.Sniffer().has_header(csvfile.read(1024))
 
+    with open(filepath, "r") as csvfile:
+        csvdata = list(csv.reader(csvfile))
     # transform into desired output
     responses = list()
-    for record in csvdata[1:]:
-        temp = list()
-        temp.append(record[0].replace('"', ""))
-        for value in record[1:]:
-            if value == "True":
-                temp.append(True)
-            elif value == "False":
-                temp.append(False)
-        responses.append(temp)
+    if has_header is True:
+        for record in csvdata[1:]:
+            temp = list()
+            temp.append(record[0].replace('"', ""))
+            for value in record[1:]:
+                if value == "True":
+                    temp.append(True)
+                elif value == "False":
+                    temp.append(False)
+            responses.append(temp)
+    else:
+        for record in csvdata:
+            temp = list()
+            temp.append(record[0].replace('"', ""))
+            for value in record[1:]:
+                if value == "True":
+                    temp.append(True)
+                elif value == "False":
+                    temp.append(False)
```

Comment:
To elaborate on my earlier suggestion of adding parsing for floats, one way of accomplishing this would be something along the lines of:
```
else:
    try:
       temp.append(float(value))
    except ValueError: # value cannot be converted to float
        pass
```

Reply:
Instead of using error handling, this could also be done with a regex string matching only floats (probably less readable imo).

Reply:
This should be handled in other PR, but not this one. The purpose of this PR just just to implement header handling.